src/Main.hs

```diff
@@ -85,6 +88,8 @@ data Opts = Opts
   { optsToken :: AuthToken
   , optsOffline :: Bool
   , optsIgnoreCase :: Bool
+  , optsSearchAll :: Bool
+  , optsShowDate :: Bool
   , optsBy :: Maybe String
   , optsOrganisation :: ParamName Organisation
   , optsFlow :: ParamName Flow
```

Comment:
It should work, as `--all` and flow name are mutually exclusive. But I'm not entirely sure how `optparse-applicative` would behave, as there is everything capturing needle, as you said - OTOH that would be a bug in the library.